advanced_animation/rendering/manim_scene.py

```
@@ -400,96 +422,180 @@ def render_scene_wrapper(scene):
+        
         try:
-            scene_content = self.generate_scene_code(storyboard_scene)
+            # Create a sandbox
+            sandbox = self.e2b_client.sandboxes.create(
+                template="python",
+                timeout_seconds=30
+            )
             
-            scene_file = self.output_dir / f"scene_{storyboard_scene.id}.py"
+            # Write code to a file in the sandbox
+            file_path = f"/tmp/code.{language}"
```

Ruff (0.14.13)

Probable insecure usage of temporary file or directory: "/tmp/code."

(S108)